### ISSUE TITLE
Cannot find module '@sanity/core/_internal'

### ISSUE THREAD

**Issue by 3200pro** [sanity-io/sanity#2919]
New Error I’ve never seen before while deploying GraphQL (sanity deploy graphql):

Error: Cannot find module '@sanity/core/_internal'

It happens when I run `sanity graphql deploy`

**So far I’ve tried:**

- Reinstalling Sanity
- Reloading node_modules and all yarn
- If I remove all of the items from my schema deploy graphql works fine. Probably because its not needing to do anything.

(Other builds I have locally seem to work fine.)

**Full Error:**
```

Error: Cannot find module '@sanity/core/_internal'
Require stack:
- <...>/config.ts
- <...>studio/schemas/documents/sitePost.js
- <...>studio/schemas/documents/index.js
- <...>studio/schemas/schema.js
- <...>studio/node_modules/@sanity/core/lib/actions/graphql/getSanitySchema.js
- <...>studio/node_modules/@sanity/core/lib/actions/graphql/deployApiAction.js
- <...>studio/node_modules/@sanity/util/lib/dynamicRequire.js
- <...>studio/node_modules/@sanity/util/lib/lazyRequire.js
- <...>studio/node_modules/@sanity/core/lib/commands/build/buildCommand.js
- <...>studio/node_modules/@sanity/core/lib/commands/index.js
- <...>studio/node_modules/@sanity/core/lib/index.js
- /usr/local/lib/node_modules/@sanity/cli/bin/sanity-cli.js
- /usr/local/lib/node_modules/@sanity/cli/bin/sanity 
```

**Dependencies:**

```
"dependencies": {
    "@sanity/base": "^2.21.9",
    "@sanity/components": "^2.14.0",
    "@sanity/core": "^2.21.8",
    "@sanity/default-layout": "^2.21.9",
    "@sanity/default-login": "^2.21.9",
    "@sanity/desk-tool": "^2.21.9",
    "@sanity/production-preview": "^2.15.0",
    "@sanity/vision": "^2.21.9",
    "prop-types": "^15.7",
    "react": "^17.0",
    "react-dom": "^17.0",
    "sanity-mobile-preview": "^1.0.8",
    "sanity-plugin-asset-source-ogimage": "^1.0.6",
    "sanity-plugin-better-slug": "^1.0.0",
    "sanity-plugin-graph-view": "^1.0.7",
    "sanity-plugin-media-library": "^2.0.1",
    "sanity-plugin-social-preview": "^0.1.5",
    "styled-components": "^5.2.0"
  },
```

Sanity Versions:

```
@sanity/cli                 2.21.7 (up to date)
@sanity/base                2.21.9 (up to date)
@sanity/components          2.14.0 (up to date)
@sanity/core                2.21.8 (up to date)
@sanity/default-layout      2.21.9 (up to date)
@sanity/default-login       2.21.9 (up to date)
@sanity/desk-tool           2.21.9 (up to date)
@sanity/production-preview  2.15.0 (up to date)
@sanity/vision              2.21.9 (up to date)
```

**What operating system are you using?**

OS

**Which versions of Node.js / npm are you running?**

7.20.6
v14.17.1

Geoff Ball mentioned that it may have to do with https://github.com/sanity-io/sanity/pull/2906

If so, I'm wondering why it's only happening to one of my installs since most use the same dependencies and version.

Thanks!

**Comment by ricokahler** [comment]
Hi @3200pro, #2903 would probably be the culprit (#2906 is actually follow up to improve reliability of #2903).

That PR provides an alternative implementation for `require.context`, which is a webpack API we try to shim out for GraphQL deploys (though we're deprecating this). That implementation imports from `@sanity/core/_internal` but this may lead to module resolution issues with certain dependencies.

Can you try either:
1. Deleting your package-lock.json file and reinstalling
2. Using something like `npm-force-resolutions` to force the exact version of `@sanity/core` which is currently `2.21.8`

```js
// package.json
{
  "name": "mysanityapp",
  "resolutions": {
    "@sanity/core": "2.21.8"
  },
  "scripts": {
    "preinstall": "npx npm-force-resolutions"
  }
}
```

To check your work, you can search your package-lock file for `@sanity/core`. In all occurrences, you should only see the version `2.21.8`.

Let me know if that doesn't work for you.

**Comment by 3200pro** [comment]
@ricokahler Thanks for the promptness!

I gave it a go and both solutions ended up at the same spot.  I checked my yarn.lock file and it looks like I'm going through the steps correctly:

```
"@sanity/core@2.21.8", "@sanity/core@^2.21.8":
  version "2.21.8"
```

End result:```
 Error: Cannot find module '@sanity/core/_internal'
Require stack:
```

**Comment by ricokahler** [comment]
@3200pro we've just released [v2.21.10](https://github.com/sanity-io/sanity/releases/tag/v2.21.10). Can you give that a try?

**Comment by 3200pro** [comment]
Everything's looking/working well here. Thanks for the heads up!
